pkg/cmd/rollbackcopy/backuputils.go

```diff
@@ -0,0 +1,132 @@
+		return fmt.Errorf("Member is NOT the leader. Returning")
+	}
+	klog.Info("Member IS the leader. Backing up")
+
+	currentClusterVersion, upgradeInProgress, err := getClusterVersionAndUpgradeInfo(cli)
+	if err != nil {
+		return fmt.Errorf("getClusterVersionAndUpgradeInfo failed: %w", err)
+	}
+
+	if upgradeInProgress {
+		return fmt.Errorf("The cluster is being upgraded. Skipping backup!")
+	}
+
+	tmpBackupDir := filepath.Join(configDir, "rollbackcopy", "tmp")
+	defer os.RemoveAll(tmpBackupDir)
+
+	if err := checkAndCreateDir(tmpBackupDir); err != nil {
+		return err
+	}
+
+	// Trying to match the output file formats with the formats of the current cluster-backup.sh script
+	dateString := time.Now().Format("2006-01-02_150405")
+	outputArchive := "static_kuberesources_" + dateString + ".tar.gz"
+	snapshotOutFile := "snapshot_" + dateString + ".db"
+
+	// Save snapshot
+	if err := saveSnapshot(cli, filepath.Join(tmpBackupDir, snapshotOutFile)); err != nil {
+		return fmt.Errorf("saveSnapshot failed: %w", err)
+	}
+
+	// Save the corresponding static pod resources
+	if err := archiveLatestResources(configDir, filepath.Join(tmpBackupDir, outputArchive)); err != nil {
+		return fmt.Errorf("archiveLatestResources failed: %w", err)
+	}
+
+	// Write the version
+	version := backupVersion{currentClusterVersion, dateString}
+	if err := putVersion(&version, tmpBackupDir); err != nil {
+		return fmt.Errorf("putVersion failed: %w", err)
+	}
+
+	if err := checkVersionsAndMoveDirs(configDir, tmpBackupDir, upgradeInProgress); err != nil {
+		return fmt.Errorf("checkVersionsAndMoveDirs failed: %w", err)
+	}
+
+	return nil
+}
+
+func checkVersionsAndMoveDirs(configDir, newBackupDir string, beingUpgraded bool) error {
+	if beingUpgraded {
+		return nil
+	}
+	currentVersionlatestDir := filepath.Join(configDir, "rollbackcopy", "currentVersion.latest")
+	currentVersionPrevDir := filepath.Join(configDir, "rollbackcopy", "currentVersion.prev")
+	if versionChanged(currentVersionlatestDir, newBackupDir) {
+		olderVersionlatestDir := filepath.Join(configDir, "rollbackcopy", "olderVersion.latest")
+		olderVersionPrevDir := filepath.Join(configDir, "rollbackcopy", "olderVersion.prev")
+		if err := safeDirRename(currentVersionPrevDir, olderVersionPrevDir, true); err != nil {
+			return err
+		}
+		if err := safeDirRename(currentVersionlatestDir, olderVersionlatestDir, true); err != nil {
+			return err
+		}
+	} else {
+		if err := safeDirRename(currentVersionlatestDir, currentVersionPrevDir, true); err != nil {
+			return err
+		}
+	}
+	if err := safeDirRename(newBackupDir, currentVersionlatestDir, false); err != nil {
+		return err
+	}
+	klog.Info("Backed up resources and snapshot to ", currentVersionlatestDir)
+	return nil
+}
+
+func checkLeadership(cli *clientv3.Client, name string) bool {
+	//return rand.Float32() < 0.5
+	flag, err := isLeader(cli, name)
+	if err != nil {
+		klog.Info("Failed to check leadership: ", err)
+		return false
```

Comment:
Would recommending returning `(bool, error)` instead so the caller can distinguish between a successful determination of true/false and a failure to make a determination (which could be retried immediately)